### DISCUSSION TITLE
Issue while Foundry Zksync

### DISCUSSION THREAD

**Discussion by audit-bykhizar** [Cyfrin/foundry-full-course-cu#3002] [Q&A] [answered]
![image_2024-12-07_023559849](https://github.com/user-attachments/assets/2a8fe606-9897-4562-86da-d40b1861f735)

**Comment by audit-bykhizar** [comment]
My password is correct; I don't know why it's still happening.

**Reply by EngrPips** [reply, marked as answer]
I am convinced you are not entering the password correctly. If you do, it won't revert with `permission denied`

**Reply by audit-bykhizar** [reply]
git clone https://github.com/matter-labs/foundry-zksync.git

bro i found out can we fix it if we use this code terminal no need password

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out.

**Comment by audit-bykhizar** [comment]
![image_2024-12-08_213027239](https://github.com/user-attachments/assets/a007d6ba-c8e4-48c1-b9db-5e34ce6c52cb)
bro i got this how can i fix that  I would really appreciate

**Reply by EngrPips** [reply]
Hello @khixarcyber, Please follow the installation guide on this page https://foundry-book.zksync.io/getting-started/installation. Let us know how it plays out.

**Comment by Aashim-Limbu** [comment]
Hey @EngrPips . I gotta a question do we have some seperate zk-sync section . Currently i'm at the NFT MarketPlace project.Just want to know as the I want to learn more on L2 .

**Reply by EngrPips** [reply]
There is no entire section that teaches only `ZKSync` as of now I believe.

**Comment by audit-bykhizar** [comment]
![image_2024-12-10_144837296](https://github.com/user-attachments/assets/12ada3a6-5be9-4908-b5de-091f26cc20ac)
Is it really necessary to no do the optional part, or should we just start with the L2 leson

**Reply by EngrPips** [reply]
I will advise you to do it but if you really don't want to do it then you can skip it.
